File: src/main/java/gg/mineral/server/entity/living/human/property/PlayerAbilities.java
package gg.mineral.server.entity.living.human.property;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.experimental.Accessors;

@AllArgsConstructor
@Data
@Accessors(fluent = true)
public class PlayerAbilities {
    private boolean isInvulnerable, flying, canFly, canInstantlyBuild, mayBuild = true;
    private float flySpeed = 0.05f, walkSpeed = 0.1f;
}

File: api/src/main/kotlin/gg/mineral/api/entity/living/human/property/PlayerAbilities.kt
package gg.mineral.api.entity.living.human.property

class PlayerAbilities(
    val isInvulnerable: Boolean = false,
    val flying: Boolean = false,
    val canFly: Boolean = false,
    val canInstantlyBuild: Boolean = false,
    val mayBuild: Boolean = true,
    val flySpeed: Float = 0.05f,
    val walkSpeed: Float = 0.1f
)
